src/preamble.js

```diff
@@ -1221,12 +1232,16 @@ if (Module['buffer']) {
 #if BINARYEN
   if (typeof WebAssembly === 'object' && typeof WebAssembly.Memory === 'function') {
 #if ASSERTIONS
-    assert(TOTAL_MEMORY % WASM_PAGE_SIZE === 0);
+    assert(TOTAL_MEMORY % PAGE_SIZE === 0);
 #endif
 #if ALLOW_MEMORY_GROWTH
-    Module['wasmMemory'] = new WebAssembly.Memory({ initial: TOTAL_MEMORY / WASM_PAGE_SIZE });
+#if BINARYEN_MEM_MAX
+    Module['wasmMemory'] = new WebAssembly.Memory({ initial: TOTAL_MEMORY / PAGE_SIZE, maximum: {{{ BINARYEN_MEM_MAX }}} / PAGE_SIZE });
```

Comment:
how about asserting that the mem max is divisible by the page size, as we do for the initial?